WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReaders/CardReaderSettingsViewModel.swift

```diff
@@ -181,7 +181,35 @@ final class CardReaderSettingsViewModel: ObservableObject {
     }
 
     /// Views can call this method to initiate a software update on the connected reader.
-    func updateSoftware() {
-        // TODO dispatch an action to update software on the connected reader
+    func checkForUpdate() {
+        let action = CardPresentPaymentAction.checkForUpdate { result in
+            switch result {
+            case .failure(let error):
+                print("===== error checking for updates ", error)
```

Comment:
We could (and should) delete these logs before merging. At this point, these can also be helpful to see progress in the console